drivers/sensor/shtcx/shtcx.c

```diff
@@ -0,0 +1,279 @@
+		return -EIO;
+	}
+
+	for (int i = 0; i < raw_len; i += (SHTCX_WORD_LEN + SHTCX_CRC8_LEN)) {
+		temp16 = sys_get_be16(&rx_buf[i]);
+		if (shtcx_compute_crc(temp16) != rx_buf[i+2]) {
+			LOG_DBG("invalid received invalid crc");
+			return -EIO;
+		}
+
+		data[dst++] = temp16;
+	}
+
+	return 0;
+}
+
+static int shtcx_sleep(const struct device *dev)
+{
+	if (shtcx_write_command(dev, SHTCX_CMD_SLEEP) < 0) {
+		return -EIO;
+	}
+	return 0;
+}
+
+static int shtcx_wakeup(const struct device *dev)
+{
+	if (shtcx_write_command(dev, SHTCX_CMD_WAKEUP)) {
+		return -EIO;
+	}
+	k_sleep(K_USEC(100));
+	return 0;
+}
+
+static int shtcx_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	struct shtcx_data *data = dev->data;
+	const struct shtcx_config *cfg = dev->config;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	if (cfg->chip == SHTC3) {
+		if (shtcx_wakeup(dev)) {
+			return -EIO;
+		}
+	}
+
+	if (shtcx_read_words(dev,
+			     measure_cmd[cfg->measure_mode][cfg->clock_stretching],
+			     (uint16_t *)&data->sample, 2,
+			     measure_wait_us[cfg->chip][cfg->measure_mode]) < 0) {
+		LOG_DBG("Failed read measurements!");
+		return -EIO;
+	}
+
+	if (cfg->chip == SHTC3) {
+		if (shtcx_sleep(dev)) {
+			LOG_DBG("Failed to initiate sleep");
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static int shtcx_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	const struct shtcx_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
+		shtcx_temperature_from_raw(data->sample.temp, val);
+	} else if (chan == SENSOR_CHAN_HUMIDITY) {
+		shtcx_humidity_from_raw(data->sample.temp, val);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api shtcx_driver_api = {
+	.sample_fetch = shtcx_sample_fetch,
+	.channel_get = shtcx_channel_get,
+};
+
+static int shtcx_init(const struct device *dev)
+{
+	const struct shtcx_config *cfg = dev->config;
+	uint16_t product_id;
+
+	if (device_is_ready(cfg->bus) ==  0) {
+		LOG_DBG("i2c bus is not ready");
+		return -ENODEV;
+	}
+
+	if (!cfg->base_address) {
+		LOG_ERR("No I2C address");
+		return -EINVAL;
+	}
+
+	k_sleep(K_USEC(SHTCX_POWER_UP_TIME_US));
+	if (cfg->chip == SHTC3) {
+		if (shtcx_wakeup(dev)) {
+			LOG_ERR("Wakeup failed");
+			return -EIO;
+		}
+	}
+
+	if (shtcx_write_command(dev, SHTCX_CMD_SOFT_RESET) < 0) {
+		LOG_ERR("soft reset failed");
+		return -EIO;
+	}
+	k_sleep(K_USEC(SHTCX_SOFT_RESET_TIME_US));
+
+	if (shtcx_read_words(dev, SHTCX_CMD_READ_ID, &product_id, 1, 0) < 0) {
+		LOG_ERR("Failed to read product id!");
+		return -EIO;
+	}
+
+	if (cfg->chip == SHTC1) {
+		if ((product_id & SHTC1_ID_MASK) != SHTC1_ID_VALUE) {
+			LOG_ERR("Device is not a SHTC1");
+			return -EINVAL;
+		}
+	} else if (cfg->chip == SHTC3) {
+		if ((product_id & SHTC3_ID_MASK) != SHTC3_ID_VALUE) {
+			LOG_ERR("Device is not a SHTC3");
+			return -EINVAL;
+		}
+		shtcx_sleep(dev);
+	} else {
+
+	}
```

Comment:
Is something missing here?

If not, the empty block should be removed.

Reply:
Usually I don't use this style, but saw #33198 fixing guide line 15.7, so I thought i could use this style.
I have updated it to use two separate if blocks.